pkg/predicate/predicate_test.go

```diff
@@ -14,6 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+//nolint:dupl
```

Comment:
Why is this needed? Can we add an explanation comment for posterity and set it on a more granular level than on the whole package?

Reply:
comments is added and level is set to single function from whole package